stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/event/handler/chat/DefaultChatEventHandler.kt

```diff
@@ -16,73 +16,129 @@
 
 package io.getstream.chat.android.offline.event.handler.chat
 
+import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.api.models.FilterObject
 import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
 import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.ChatEvent
+import io.getstream.chat.android.client.events.CidEvent
+import io.getstream.chat.android.client.events.HasChannel
 import io.getstream.chat.android.client.events.MemberAddedEvent
 import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NewMessageEvent
 import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
 import io.getstream.chat.android.client.events.NotificationMessageNewEvent
 import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
 import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Member
 import kotlinx.coroutines.flow.StateFlow
 
 /**
- * Default implementation of [ChatEventHandler] which covers the default filter of channels.
- * The channel will be added if the current user is a member and it will be removed otherwise.
+ * Default implementation of [ChatEventHandler] which is based on the current user membership.
  *
  * @param channels The list of visible channels.
+ * @param chatClient The client used to communicate to the API.
  */
-public class DefaultChatEventHandler(
+public open class DefaultChatEventHandler(
     private val channels: StateFlow<List<Channel>?>,
+    private val chatClient: ChatClient = ChatClient.instance(),
```

Comment:
Let's use `ClientState` or `StateFlow<User>` instead of `ChatClient`. We should avoid to use classes that are too broad like `ChatClient`, it is better to use classes that provide exactly what you need.

Reply:
good point